sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1763,6 +1764,39 @@ def test_penalty_none(solver):
     )
 
 
+@pytest.mark.parametrize(
+    "params",
+    [{'penalty': 'l1', 'dual': False, 'tol': 1e-12, 'max_iter': 1000},
+     {'penalty': 'l2', 'dual': True, 'tol': 1e-12, 'max_iter': 1000},
+     {'penalty': 'l2', 'dual': False, 'tol': 1e-12, 'max_iter': 1000}]
+)
+def test_logisticregression_liblinear_sample_weight(params):
+    # check that we support sample_weight with liblinear in all possible cases:
+    # l1-primal, l2-primal, l2-dual
+    X = np.array([[1, 3], [1, 3], [1, 3], [1, 3],
+                  [2, 1], [2, 1], [2, 1], [2, 1],
+                  [3, 3], [3, 3], [3, 3], [3, 3],
+                  [4, 1], [4, 1], [4, 1], [4, 1]], dtype=np.dtype('float'))
+    y = np.array([1, 1, 1, 1, 2, 2, 2, 2,
+                  1, 1, 1, 1, 2, 2, 2, 2], dtype=np.dtype('int'))
+
+    X2 = np.vstack([X, X])
+    y2 = np.hstack([y, 3 - y])
+    sample_weight = np.ones(shape=len(y) * 2)
+    sample_weight[len(y):] = 0
+    X2, y2, sample_weight = shuffle(X2, y2, sample_weight, random_state=0)
+
+    base_clf = LogisticRegression(solver='liblinear', random_state=42)
+    base_clf.set_params(**params)
+    clf_no_weight = clone(base_clf).fit(X, y)
+    clf_with_weight = clone(base_clf).fit(X2, y2, sample_weight=sample_weight)
+
+    for method in ("predict", "predict_proba", "decision_function"):
+        X_clf_no_weight = getattr(clf_no_weight, method)(X)
+        X_clf_with_weight = getattr(clf_with_weight, method)(X)
+        assert_allclose(X_clf_no_weight, X_clf_with_weight)
```

Comment:
don't we prefer `assert np.allclose`?

Reply:
uhm no. `assert_allclose` are part of the numpy testing tools:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_allclose.html

when using scalar, in this case we use `pytest.approx`

Reply:
what we prefer is `assert_allclose` instead of `assert_array_almost_equal` as well.